README.md

```diff
@@ -297,7 +295,52 @@ sub2.setTimeout(1000);
 ```
 
 
-Message payloads can be strings, binary, or json
+Normally unsubscribe will cancel a subscription by sending a command to the server
+and then cancelling the message handler for the subscription. In such cases it is
+possible for messages to be pending processing to be dropped.
+
+To orderly unsubscribe, `drain()` sends the unsubscribe to the server and
+flushes with a handler that that cancels the subscription. Because the cancel
+happens when the flush completes, drain ensures that the server will not send
+any additional messages to the subscription and that all sent messages have
+been processed by the client. 
+
+This functionality is particularly useful to queue subscribers.
+
+```typescript
+let sub3 = await nc.subscribe('foo', (err, msg) => {
+    // do something
+});
+// sometime in the future drain the subscription. Drain returns a promise
+// when the drain promise resolves, the subscription finished processing 
+// all pending inbound messages for the subscription.
+let p = await sub3.drain();
+```
+
+Similarly to subscription `drain()`, an entire connection can be drained.
+
+```typescript
+let nc10 = await connect();
+
+// create subscriptions etc
+...
+
+// When drain is called:
+// - no additional subscriptions can be created.
+// - all open subscriptions drain (excepting the global request/reply)
+// When all drain requests resolve:
+// - a flush is sent to the server to drain outbound messages
+// - publishing or performing request reply results in an error
+// - only action for the client is to close the connection
+await nc10.drain();
+
+// when drain resolves, client must close - no further interaction with
+// the library is possible.
+nc10.close();
```

Comment:
Ok, that's a departure from the Go client, but there may be reasons. I will see when looking at the rest.